substrate/frame/contracts/fixtures/build.rs

```diff
@@ -216,6 +211,65 @@ fn post_process_wasm(input_path: &Path, output_path: &Path) -> Result<()> {
 	serialize_to_file(output_path, module).map_err(Into::into)
 }
 
+/// Check if RISC-V toolchain is installed.
+fn is_riscv_supported() -> bool {
+	use std::sync::OnceLock;
+	static IS_RISCV_SUPPORTED: OnceLock<bool> = OnceLock::new();
+	*IS_RISCV_SUPPORTED.get_or_init(|| {
+		let cmd = Command::new("rustup")
+			.args(&["toolchain", "list"])
+			.output()
+			.expect("failed to execute process");
+
+		let stdout = String::from_utf8(cmd.stdout).unwrap();
+		stdout.contains("riscv32em-nightly-2023-04-05")
+	})
+}
```

Comment:
We don't want to silently fail if the toolchain is not installed. How would be know if our tests are okay? This seems very dangerous.

Reply:
Makes sense.

Reply:
Why can't we install the toolchain in our macOS image?